### DISCUSSION TITLE
Improve pybind

### DISCUSSION THREAD

**Discussion by Kuree** [MikePopoloski/slang#678] [Ideas]
Currently when `pyslang` is used with IDE such as VSCode or PyCharm, the function signature provided by pybind is not very informative. I'm wondering if you'd like some PR to enhance the pybind code.

Here is some incomplete patch I have: https://github.com/MikePopoloski/slang/compare/master...Kuree:pybind-arg?expand=1

It aims to improve two things:
1. Give function arguments more meaningful names. By default, if `py::arg` is not provided, `arg0` is used.
2. Add back default argument values. This should make Python API similar to that of C++.

Please let me know what you think. I'm happy to finish up the rest of refactoring.

**Comment by MikePopoloski** [comment]
Of course yes, I would take PRs that improve the bindings. I don't have time to do that kind of work myself but it does make the bindings unambiguously better.
